src/app/mapControls/Filter.js

```diff
@@ -0,0 +1,227 @@
+        },
+        itemClicked: function (value) {
+            // summary:
+            //      description
+            // value: String
+            //      value of the item that was clicked
+            console.log('app/mapControls/Filter:itemClicked', arguments);
+
+            var index = this.selectedValues.indexOf(value);
+            if (index === -1) {
+                this.selectedValues.push(value);
+            } else {
+                this.selectedValues.splice(index, 1);
+            }
+
+            this.numSpan.innerHTML = this.selectedValues.length;
+
+            this.emit('changed');
+        },
+        getQuery: function () {
+            // summary:
+            //      assembles all selected values into a def query
+            console.log('app/mapControls/Filter:getQuery', arguments);
+
+            if (this.selectedValues.length) {
+                var values;
+                if (this.fieldType === c.TYPE_TEXT) {
+                    values = this.selectedValues.map(function (v) {
+                        return "'" + v + "'";
+                    });
+                } else {
+                    values = this.selectedValues;
+                }
+                if (this.any) {
+                    var where = this.fieldName + ' IN (' + values.join(', ') + ')';
+                    return this.getRelatedTableQuery(where);
+                } else {
+                    var that = this;
+                    return values.reduce(function (previousReturn, currentValue) {
+                        var where = that.fieldName + ' = ' + currentValue;
+                        if (!previousReturn) {
+                            return that.getRelatedTableQuery(where);
+                        } else {
+                            return previousReturn + ' AND ' + that.getRelatedTableQuery(where);
+                        }
+                    }, false);
+                }
+            } else {
+                return undefined;
+            }
+        },
+        toggleAny: function () {
+            // summary:
+            //      description
+            console.log('app/mapControls/Filter:toggleAny', arguments);
+
+            var that = this;
+            setTimeout(function () {
+                that.any = domClass.contains(that.anyBtn, 'active');
+                that.emit('changed');
+            }, 0);
+        },
+        remove: function () {
+            // summary:
+            //      removes widget from visible filters
+            console.log('app/mapControls/Filter:remove', arguments);
+
+            this.clear();
+
+            this.emit('removed', this);
+        },
+        open: function () {
+            // summary:
+            //      opens the body of the filter
+            console.log('app/mapControls/Filter:open', arguments);
+
+            $(this.body).collapse('show');
+
+            this.inherited(arguments);
+        },
+        getRelatedTableQuery: function (where) {
```

Comment:
Ok, I think this is ok. We will need to disable the standardized queries on the server since the sub select will fail that test. 
> Steps:
Open the Administrator Directory and log in with a user that has administrative permissions to your site. The Administrator Directory is typically available at http://gisserver.domain.com:6080/arcgis/admin.
Click system > properties > update.
On the Operation - update page, enter the following string into the System Properties dialog box:
{"standardizedQueries": "false"}
Click Update.
Restart ArcGIS Server.

The point line and poly tables do carry a TypeDescription and Project_ID value but you would have to query each separately which I do not like.